src/log.cc

```diff
@@ -139,33 +143,34 @@ save_log (char *uri, void *data)
       return;
     }
 
-  /* XXX - Using gnome_vfs_create with exclusive == true to check for
-           file existence doesn't work with obex.  Why am I not
-           surprised?
-   */
+  file = g_file_new_for_uri (uri);
+  info = g_file_query_info (file, G_FILE_ATTRIBUTE_ACCESS_CAN_WRITE,
+                            G_FILE_QUERY_INFO_NONE, NULL, &error);
 
-  result = gnome_vfs_get_file_info (uri, &info, GNOME_VFS_FILE_INFO_DEFAULT);
-  if (result == GNOME_VFS_OK)
+  if (info)
     {
       ask_custom (dgettext ("hildon-fm", "docm_nc_replace_file"),
 		  dgettext ("hildon-libs", "wdgt_bd_yes"),
 		  dgettext ("hildon-libs", "wdgt_bd_no"),
 		  save_log_cont, uri);
     }
-  else if (result != GNOME_VFS_ERROR_NOT_FOUND)
+  else if (!g_error_matches (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
     {
       /* Remove the previously stored dir if failed */
       if (last_save_log_dir != NULL)
         g_free (last_save_log_dir);
       last_save_log_dir = NULL;
 
       /* Annoy user with error */
-      annoy_user_with_gnome_vfs_result (result, uri,
-					save_log_do_nothing, NULL);
+      annoy_user_with_gerror (error, uri, save_log_do_nothing, NULL);
```

Comment:
Maybe change it to annoy_user_with_error?

Reply:
I wouldn't, since there are multiple `annoy_user` functions, and I'd rather keep it explicit.

Reply:
Ok, then, how about doing annoy_user_with_g_error, to be consitent with how glib/gtk+ does with its functions/variables naming?

There're some rules in translating mixed upper&lower case names to only lowercase.
In simple, it looks like this:
HelloWorld -> hello_world, GObject -> g_object, ViewerFile -> viewer_file, etc.
Take a look here for some examples: https://developer.gnome.org/gobject/stable/gtype-conventions.html

In our case it would be:
GError -> g_error

Reply:
Back in history, when h-a-m used to report errors with GError, the function was called `annoy_user_with_gerror` and I've kept it the same.

Reply:
As far as I can see annoy_user_with_gerror() was already declared in file src/instr.cc (as static) before your commits. What you did is overlapping this function with replaced one. BTW, src/instr.cc includes the file <util.h> where you declared "new" annoy_user_with_gerror() function. I may be wrong, but all this just seems a bit wrong to me.

P.S. What "back in history" commit did you mean?

This is from src/instr.cc:
```
static void
annoy_user_with_gerror (const char *filename, GError *error,
                        void (*cont) (void *), void *data)
```
This is from src/util.h (please, fix the indentation inconsistency you introduced):
```
void annoy_user_with_gerror (GError *error,
                                       const gchar *detail,
                                       void (*cont) (void *), void *data);
```